### ISSUE TITLE
document difference from `format_args!`

### ISSUE THREAD

**Issue by matklad** [Lucretiel/lazy_format#7]
From the description, it sounds like this does the same thing as `format_args`. I _think_ the difference is that `format_args` borrows the arguments, while `lazy_format` moves them, and avoids lifetimes. Would be helpful to exaplain that in the docs!

**Comment by Lucretiel** [comment]
Correct; this library was created specifically because `format_args` only ever borrows its arguments and is therefore difficult to compose (for instance, you can never usefully use it in `iterator.map`). This is a good idea.
